### ISSUE TITLE
GLIBCXX_3.4.20 not found

### ISSUE THREAD

**Issue by ASoleimaniB** [allenai/longformer#23]
When I run the model I get "`GLIBCXX_3.4.20' not found" error! Do you have any ideas?

> OSError: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/myuser/anaconda3/envs/longformer/lib/python3.7/site-packages/tvm/libtvm_runtime.so)

**Comment by leej35** [comment]
I had the same issue and just resolved it. Please note that my solution could not be right answer for your situation. 

TLDR: it seems the issue could be caused by incompatible compiler's version to the compiled library that Longformer uses. You could resolve it through (1) install new library (glibc) through condo and then (2) add conda's library path to your environment.    

**Part 1. Check your gcc version**
In my case, I first checked my compiler (gcc) version:
```
gcc -v
```
With this, I got to know that my compiler's version is `gcc version 4.8.5`.

Through this table on the link, I found 4.8.5 uses/support GLIBCXX_3.4.19, not the one that Longformer needs. (Link: https://github.com/FoldingAtHome/fah-issues/issues/1147#issuecomment-134731211)
The link also says GLIBCXX_3.4.20 is indeed on gcc version 4.9.0.  So if your gcc version is below 4.9.0, it could be the same reason that I have.


**Part 2. Install libgcc through conda**
Based on my search on google, it seems basically you need to upgrade gcc but it requires sudo. For my case I don't have sudo. Another way to install the newer library is through conda's libgcc. Run following line:
```
conda install libgcc
```

**Part 3. Add conda's library path to your LD_LIBRARY_PATH env variable.** 
```
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/path/to/your/anaconda3/lib
```

Then, the Longformer works as charm!

**Comment by ibeltagy** [comment]
Closing. Please feel free to reopen or create a new one if you have more questions.
